controllers/storageclient_controller.go

```diff
@@ -206,6 +230,19 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileClusterResourceQuota(spec *quotav1.ClusterResourceQuotaSpec) error {
+	clusterResourceQuota := &quotav1.ClusterResourceQuota{}
+	clusterResourceQuota.Name = fmt.Sprintf("%s-%s", r.storageClient.Name, "quota")
+	_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, clusterResourceQuota, func() error {
+		clusterResourceQuota.Spec = *spec
+		return r.own(clusterResourceQuota)
```

Comment:
As a convention, we own first. Mainly because we don't want to run all of the update code (in the general use case) if ownership failed. 
```suggestion
		if err := r.own(clusterResourceQuota); err != nil {
			return err
		}

		clusterResourceQuota.Spec = *spec
		return nil 
```